homeassistant/components/nws/__init__.py

```diff
@@ -1 +1,186 @@
-"""NWS Integration."""
+"""The National Weather Service integration."""
+import asyncio
+import datetime
+import logging
+
+import aiohttp
+from pynws import SimpleNWS
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_SCAN_INTERVAL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import discovery
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import CONF_STATION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+_INDIVIDUAL_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Inclusive(
+            CONF_LATITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.latitude,
+        vol.Inclusive(
+            CONF_LONGITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.longitude,
+        vol.Optional(CONF_SCAN_INTERVAL): vol.All(cv.positive_int, vol.Range(min=5)),
+        vol.Optional(CONF_STATION): cv.string,
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.All(cv.ensure_list, [_INDIVIDUAL_SCHEMA])}, extra=vol.ALLOW_EXTRA,
+)
+
+PLATFORMS = ["weather"]
+
+
+def unique_id(latitude, longitude):
+    """Return unique id for entries in configuration."""
+    return f"{DOMAIN}_{latitude}_{longitude}"
```

Comment:
AFAICT, unique ids are per platform.   You should be able to drop DOMAIN



`helpers/entity.py`

```

    def __eq__(self, other):
        """Return the comparison."""
        if not isinstance(other, self.__class__):
            return False

        # Can only decide equality if both have a unique id
        if self.unique_id is None or other.unique_id is None:
            return False

        # Ensure they belong to the same platform
        if self.platform is not None or other.platform is not None:
            if self.platform is None or other.platform is None:
                return False

            if self.platform.platform != other.platform.platform:
                return False

        return self.unique_id == other.unique_id
```

Reply:
Yes, the entity registry, which handles the entity unique ids, is aware of both the integration domain and platform domain of an entity.

Reply:
Thanks for the detailed info.  I was under the impression that I had to provide a totally unique id across the entire ecosystem, but this makes much more sense.  I was wondering how I would handle the addition of more platforms, but this makes it clear to me.